sos/report/plugins/proxmox.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2024 ORNESS/Ditrit Drien Breton <drien.breton@orness.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import json
+from pathlib import Path
+
+from sos.report.plugins import Plugin, DebianPlugin, PluginOpt
+
+
+class Proxmox(Plugin, DebianPlugin):
+    """
+    This plugin will capture information about the system's
+    Proxmox Virtualization Environment.
+
+    It will collect information about the cluster, nodes, pools and
+    storage from the Proxmox API.
+    """
+    short_desc = 'Proxmox cluster information'
+
+    plugin_name = 'proxmox'
+    packages = ('proxmox-ve',)
+
+    option_list = [
+        PluginOpt('output_formats',
+                  desc='List of output formats to use '
+                       'for the commands separated by ":".',
+                  default='text',
+                  ),
+        PluginOpt('cmd_override_file',
+                  desc='File containing the commands '
+                       'to run instead of the default ones.',
+                  default=None,
+                  val_type=[None, Path, str],
+                  ),
+    ]
+
+    def setup(self):
+        output_formats = self.get_option('output_formats').split(':')
+
+        commands = [
+            'cluster/resources',
+            'cluster/config/nodes',
+            'cluster/options',
+            'nodes/:id/status',
+            'nodes/:id/storage',
+            'nodes/:id/network',
+            'pools',
+            'storage',
+            'storage/:sid',
+            'cluster/ceph/status'
+            'cluster/ceph/metadata'
+            'cluster/ceph/flags'
+        ]
+
+        cmd_paths = []
+
+        override_file = self.get_option('cmd_override_file')
+        if override_file is not None:
+            with open(override_file, 'r') as f:
+                for line in f:
+                    cmd_paths.extend(self.build_cmd_paths(line.strip()))
+        else:
+            for command in commands:
+                cmd_paths.extend(self.build_cmd_paths(command))
+
+        for cmd_path in cmd_paths:
+            for output_format in output_formats:
+                cmd = f'pvesh get {cmd_path} --output-format {output_format}'
+                file_name = cmd.replace("/", ".").replace(" ", "_")
```

Comment:
We do this internally in our command-to-file name mangling, so there is no need to do this yourself or to use `suggest_filename` here for it.